TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -1783,4 +1790,27 @@ private static DataDefinition GetDataDefinitionType(Node node, Node symbol, bool
         }
 
     }
+
+    static class RepositoryChecker
+    {
+        // Intrinsic functions which are not allowed in the REPOSITORY paragraph
+        private static readonly HashSet<string> _NotAllowedIntrinsicFunctions = new(StringComparer.OrdinalIgnoreCase){ "WHEN-COMPILED" };
+
+        public static void OnNode([NotNull] Repository repository)
```

Comment:
In fact this whole check can be moved to the CodeElement phase.

Reply:
Done

Reply:
Move to a new `CodeElementChecker`